Sources/ScipioKit/DescriptionPackage.swift

```diff
@@ -66,12 +66,21 @@ struct DescriptionPackage: PackageLocator {
     /// Then, use package versions only from existing Package.resolved.
     ///   If it is `true`, Package.resolved never be updated.
     ///   Instead, the resolving will fail if the Package.resolved is mis-matched with the workspace.
-    init(packageDirectory: ScipioAbsolutePath, mode: Runner.Mode, onlyUseVersionsFromResolvedFile: Bool) throws {
+    init(
+        packageDirectory: ScipioAbsolutePath,
+        mode: Runner.Mode,
+        onlyUseVersionsFromResolvedFile: Bool,
+        toolchainEnvironment: ToolchainEnvironment? = nil
+    ) throws {
         self.packageDirectory = packageDirectory
         self.mode = mode
-
-        let toolchain = try UserToolchain(swiftSDK: try .hostSwiftSDK())
-        self.toolchain = toolchain
+        self.toolchain = try UserToolchain(
+            swiftSDK: try .hostSwiftSDK(
+                toolchainEnvironment?.toolchainBinPath,
+                environment: toolchainEnvironment.asSwiftPMEnvironment
+            ),
+            environment: toolchainEnvironment.asSwiftPMEnvironment
+        )
```

Comment:
Sorry for the belated response. I'm back as of today! 

We’re almost finished with our Scipio-powered tool, and during final testing, we discovered another environment-related issue (hopefully the last one!).

There’s a bug in SwiftPM where Environment isn’t passed down to the shell-executing function [here](https://github.com/swiftlang/swift-package-manager/blob/42aaad6d4b9cd5e18112412d346c054f1a8f46a7/Sources/PackageModel/UserToolchain.swift#L260-L265). Fortunately, it’s possible to work around this by supplying a custom binDir (pointing to the bin dir of the current toolchain in our case) [a few lines above](https://github.com/swiftlang/swift-package-manager/blob/42aaad6d4b9cd5e18112412d346c054f1a8f46a7/Sources/PackageModel/UserToolchain.swift#L255-L256).

I’m planning to file a fix for this in SwiftPM soon. Although it’s not a critical issue, resolving it would allow a cleaner setup by removing the need to specify a custom bin path.

Apologies for including this change in the current PR, but I believe it’s closely related to its focus, and it’s purely additive, leaving existing behavior unaffected.

As mentioned, this appears to be the last environment-related enhancement needed to enable Scipio for multi-toolchain use. 🎉